File: 2136. Earliest Possible Day of Full Bloom/Solution.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.stream.IntStream;

class Solution {
    public int earliestFullBloom_MK1(int[] plantTime, int[] growTime) {
        Integer[] indices = IntStream.range(0, plantTime.length).boxed().toArray(Integer[]::new);
        Arrays.sort(indices, Comparator.comparingInt(i -> growTime[i]));
        int res = 0;
        for (Integer i : indices) {
            res = Math.max(res, growTime[i]);
            res += plantTime[i];
        }
        return res;
    }

    public int earliestFullBloom_MK2(int[] plantTime, int[] growTime) {
        Integer[] indices = IntStream.range(0, plantTime.length).boxed().toArray(Integer[]::new);
        Arrays.sort(indices, (i, j) -> growTime[j] - growTime[i]);
        int res = 0, endTime = 0;
        for (Integer i : indices) {
            endTime += plantTime[i];
            res = Math.max(res, endTime + growTime[i]);
        }
        return res;
    }
}

File: 2136. Earliest Possible Day of Full Bloom/Solution.py
from typing import List


class Solution:
    def earliestFullBloom_MK1(self, plantTime: List[int], growTime: List[int]) -> int:
        flowers = sorted(zip(plantTime, growTime), key=lambda x: x[1])
        res = 0
        for pt, gt in flowers:
            res = max(res, gt)
            res += pt
        return res

    def earliestFullBloom_MK2(self, plantTime: List[int], growTime: List[int]) -> int:
        flowers = sorted(zip(plantTime, growTime), key=lambda x: x[0], reverse=True)
        res, t = 0, 0
        for pt, gt in flowers:
            res = max(res, t + pt)
            res += gt
        return res
